### PR TITLE
Issue #270 - Add Sklearn Matrix Decomposition Operators

### LINKED ISSUE #270: Add Sklearn Matrix Decomposition Operators
Add support for the following operators

- PCA
- KernelPCA
- TruncatedSVD
- FastICA

### PR DESCRIPTION
**Pull request by scnakandala**
Add support for the following operators

PCA
KernelPCA
TruncatedSVD
FastICA

Testing:
- I added a new test file test_sklearn_decomposition.py to test various matrix decomposition operators.

### CODE REVIEW

**Comment by scnakandala** [comment]
KernelPCA fails with kernels other than linear. Working on fixing them.

**Comment by scnakandala** [comment]
Added more kernel support for KernelPCA including and fixed the issues with existing ones.
Now it supports linear, rbf, poly, sigmoid, cosine, and precomputed kernels.

File: tests/test_sklearn_decomposition.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    # KernelPCA cosine kernel
+    def test_kernel_pca_converter_cosine(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="cosine"))
+
+    # KernelPCA precomputed kernel
+    def test_kernel_pca_converter_precomputed(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="precomputed"), precompute=True)
+
+    # TODO: Fails on macos-latest Python 3.8 due to a sklearn bug.
+    # FastICA converter with n_components none
```
Comment (ksaur) [inline tests/test_sklearn_decomposition.py:105]: Can you please open an issue for this so we can fix it up when the bug is fixed?

**Review by ksaur — APPROVED** [review]

File: tests/test_sklearn_decomposition.py (hunk 1 above)
Comment (scnakandala) [reply tests/test_sklearn_decomposition.py:105]: done

**Pull request merged**